### ISSUE TITLE
Error in Longformer attention mask using apex mixed precision

### ISSUE THREAD

**Issue by peskotivesgeroff** [huggingface/transformers#4525]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Longformer

Language I am using the model on (English, Chinese ...): English

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

1. Install latest transformers (2.10.0) and apex (0.1)
2. Code: 
```
import torch
from transformers import LongformerTokenizer, LongformerModel, LongformerConfig
from apex import amp


tokenizer = LongformerTokenizer.from_pretrained('longformer-base-4096')
config = LongformerConfig.from_pretrained('longformer-base-4096')
model = LongformerModel.from_pretrained('longformer-base-4096').cuda()
optimizer = torch.optim.Adam(model.parameters(), lr=1e-5)

model, optimizer = amp.initialize(model, optimizer, opt_level='O1')

# toy input example
inputs = torch.randint(config.vocab_size, (1, 1024)).cuda() # randomly select tokens with sequence length 1024
mask = torch.ones(1, 1024).cuda() # set mask for every token to local attention
mask[0] = 2. # global attention for first token

outputs = model(inputs, attention_mask=mask)
```

## Expected behavior

Error message:
```
File "/home/miniconda3/envs/test_transformer/lib/python3.8/site-packages/transformers/modeling_longformer.py", line 374, in
forward
    attn[extra_attention_mask_nonzeros[::-1]] = nonzero_selected_attn.view(
RuntimeError: expected dtype Half but got dtype Float
```
`attn` is half precision but is assigned a tensor that is casted into single precision `.type_as(hidden_states)` in line 376 .

## Environment info

     
- `transformers` version: 2.10.0
- Platform: Linux-4.15.0-65-generic-x86_64-with-glibc2.10
- Python version: 3.8.3
- PyTorch version (GPU?): 1.5.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

**Comment by BramVanroy** [comment]
Good catch! I guess that soon you'd want to move over PyTorch's built-in AMP which takes care of this automatically (I _think_), but for the time being your suggestion is a good fix. You can submit a PR if you want!

**Comment by patrickvonplaten** [comment]
I think this is solved with the PR #4574 no?
